src/subscript-loc.c

```diff
@@ -217,8 +254,13 @@ r_obj* int_invert_location(r_obj* subscript,
     int j = *data;
 
     if (j == r_globals.na_int) {
-      stop_location_negative_missing(subscript, opts);
+      switch (opts->missing) {
+      case SUBSCRIPT_MISSING_PROPAGATE: stop_location_negative_missing(subscript, opts);
+      case SUBSCRIPT_MISSING_REMOVE: continue;
```

Comment:
As discussed:

```r
# remains an error
num_as_location(c(-1, NA), n = 2, missing = "propagate", negative = "invert")

# this is allowed
num_as_location(c(-1, NA), n = 2, missing = "remove", negative = "invert")
```

Reply:
Add a comment that we're just following base R here? And that it does not really make sense to invert an `NA` value so we can't meaningfully propagate, and we'd rather not drop missing values where base R doesn't.